cartographer/io/submap_painter.h

```diff
@@ -55,10 +58,43 @@ struct SubmapSlice {
   int metadata_version = -1;
 };
 
+struct SubmapTexture {
+  struct Pixels {
+    std::vector<char> intensity;
+    std::vector<char> alpha;
+  };
+  Pixels pixels;
+  int width;
+  int height;
+  double resolution;
+  ::cartographer::transform::Rigid3d slice_pose;
+};
+
+struct SubmapTextures {
+  int version;
+  std::vector<SubmapTexture> textures;
+};
+
 PaintSubmapSlicesResult PaintSubmapSlices(
     const std::map<::cartographer::mapping::SubmapId, SubmapSlice>& submaps,
     double resolution);
 
+void FillSubmapSlice(
+    const ::cartographer::transform::Rigid3d& global_submap_pose,
+    const ::cartographer::mapping::proto::Submap& proto,
+    SubmapSlice* const submap_slice);
+
+// Unpacks cell data as provided by the backend into 'intensity' and 'alpha'.
+SubmapTexture::Pixels UnpackTextureData(const std::string& compressed_cells,
+                                        int width, int height);
+
```

Comment:
In .cc file you have here 'const int width, const int height'. Same goes for DrawTexture(). Please, correct for consistency. Just remove "const"s, you are passing by value anyways.